parsl/executors/base.py

```diff
@@ -41,7 +41,7 @@ def start(self) -> None:
         pass
 
     @abstractmethod
-    def submit(self, func: Callable, *args: Any, **kwargs: Any) -> Future:
+    def submit(self, func: Callable, resource_specification: Dict, *args: Any, **kwargs: Any) -> Future:
```

Comment:
Is there more description of what goes in `resource_specification` than a plain Dict? For example, is it a dictionary of strings -> anything? (`Dict[str,Any]`) or strings -> strings? (`Dict[str,str]`)